src/hyperlight_host/src/sandbox/outb.rs

```diff
@@ -136,23 +118,31 @@ fn handle_outb_impl(
             Ok(())
         }
         OutBAction::Abort => {
+            let byte = u64::from(data[0]);
             let guest_error = ErrorCode::from(byte);
-            let panic_context = mem_mgr.as_mut().read_guest_panic_context_data()?;
-            // trim off trailing \0 bytes if they exist
-            let index_opt = panic_context.iter().position(|&x| x == 0x00);
-            let trimmed = match index_opt {
-                Some(n) => &panic_context[0..n],
-                None => &panic_context,
-            };
-            let s = String::from_utf8_lossy(trimmed);
+
             match guest_error {
                 ErrorCode::StackOverflow => Err(HyperlightError::StackOverflow()),
-                _ => Err(HyperlightError::GuestAborted(
-                    byte as u8,
-                    s.trim().to_string(),
-                )),
+                _ => {
+                    let message = match data.get(1) {
+                        Some(&exception_code) => match Exception::try_from(exception_code) {
+                            Ok(exception) => format!("Exception: {:?}", exception),
+                            Err(e) => {
+                                format!("Unknown exception code: {:#x} ({})", exception_code, e)
+                            }
+                        },
+                        None => "See stderr for panic context".into(),
+                    };
+
+                    Err(HyperlightError::GuestAborted(byte as u8, message))
+                }
             }
         }
+        OutBAction::DebugPrint => {
+            let s = String::from_utf8_lossy(&data);
+            eprint!("{}", s);
```

Comment:
should we not append a new line at the end of the data, it seems like this is going to potentially concatenate a lot of data to stderr? Given you are padding with 0x7f then we could replace that with a new line although you would have to send an extra 4 bytes in the case that the string length was divisible by 4.

Also I think it might be nice to allow the host to override or alternatively maybe we should just use error!() here , which would mean gathering the string into a buffer until you get the final 0x7f?

Reply:
We send 4 bytes at a time. If we added a new line, we'd get print output that looks like this:
```
hell
o, w
orld
```

I chose 0x7f for padding because it is a non-printable character. A new-line is 0x0A, which is the same value as the `InvalidTSS` exception. I considered using a macro from the log crate, but it felt like a bit of overengineering for just `debug_print`. Do we want to allow people to customize an `OutBAction`?